src/lib/OS/SyscallHandler.cc

```diff
@@ -488,20 +494,60 @@ void SyscallHandler::handleSyscall() {
         if (bitmask != 1) {
           return concludeSyscall({}, true);
         }
-        uint64_t retval = (pid == 0) ? 1 : 0;
-        stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {retval}};
-        stateChange.memoryAddresses.push_back({mask, 1});
+        uint64_t retval = static_cast<uint64_t>(bitmask);
+        stateChange = {
+            ChangeType::REPLACEMENT, {currentInfo_.ret}, {sizeof(retval)}};
+        stateChange.memoryAddresses.push_back({mask, 8});
         stateChange.memoryAddressValues.push_back(bitmask);
       } else {
         stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {-1ll}};
       }
       break;
     }
+    case 124: {  // sched_yield
+      // Non args passed in
+      // Have core go to idle after syscall, forcing the current Process to be
+      // de-scheduled
+      return concludeSyscall({}, false, true);
```

Comment:
Should this not have a default return value of 0?

Reply:
Updated